app/routes/_mainLayout.t7_.$character.$move.tsx

```diff
@@ -40,38 +42,48 @@ export const loader = async ({ params }: DataFunctionArgs) => {
     });
 
     if (!response.data.values) {
-      throw json("not found", { status: 401, statusText: "Not found 1" });
```

Comment:
I think we can just keep using the json util for this? makes it easer to add more info in the future?
throw json("not found", {
        status: ServerStatusCode.NotFound,
        statusText: "Not found 2",
      }

Reply:
Great suggestion! I was not sure what was preferred within the project. If we keep it as JSON we then need to do something like this within the `catch`. 

```typescript 
   catch (error) {
      const typedError = error as OurCustomErrorType;
      const status = typedError.status || ServerErrorCode.ServerError;
      const statusText = typedError.status || 'server error';
      throw new Response(null, { status, statusText })
   }
```

The `ErrorResponse` interface from Remix implements the following interface: 

```typescript
interface ErrorReponse {
   status: string;
   statusText: string;
   data: any;
}
```

A great solution could be always to throw the `ErrorResponse` to keep the interface standardized and have our extra/own properties within the `data: any` property. 🤔 

I'm not an expert in Remix and I'm not sure that is preferred due to ErrorHandling, but I think it could be smart to ensure a consistent interface for error handling.

Reply:
I didn't notice that I was already mixing new Response and json(). I think the solution in the lastest commit is great :)